dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/chart/impl/DefaultChartService.java

```diff
@@ -796,7 +798,9 @@ else if ( chart.isAnalyticsType( AnalyticsType.EVENT ) )
         valueMap = DimensionalObjectUtils.getSortedKeysMap( valueMap );
 
         List<NameableObject> seriez = new ArrayList<>( chart.series() );
-        List<NameableObject> categories = new ArrayList<>( chart.category() );
+
+        List<DimensionalItemObject> dimensionalItemObjects = chart.category() != null ? chart.category() : emptyList();

```

Comment:
Could we do:

`List<NameableObject> categories = new ArrayList<>( ObjectUtils.firstNonNull( chart.category(), new ArrayList<>() ) );

Reply:
Thanks Lars. I applied the same code I applied for other branches just to keep them in sync. It's pretty much the same in the end.